### ISSUE TITLE
Updated RPi make instructions

### ISSUE THREAD

**Issue by anandmg101** [viktorgino/headunit-desktop#33]
Hi Viktorgino, thanks for fixing qml_gui folder thing so quickly! One more request. Its seems there has been few changes in the application. So the final app file is different I believe. Instead of "app.pro", I believe it is "headunit-desktop.pro"
So the final command "make -j4 sub-app-pro" gives me a program without AA. I did try "make -j4 sub-headunit-desktop-pro" but getting error that there is no make config file.

Can you help me with that please?

**Comment by pawel0987** [comment]
Hello,
I think I have similar problem. Building instruction is outdated or not complete (I have tried http://headunit.viktorgino.me). I stuck on compiling protobuf files. The error is `headunit/hu/hu.proto: No such file or directory`. When skipping this step and building with QTCreator, there are another errors like: `Unknown module(s) in QT: BluezQt` (I have installed everything I could for bluez including qt-connectivity).
Maybe for people who knows QT it is simple and does not require clarification but I have not idea how to build and run the project. I hope someone can write working build documentation so I will be able to try this software and maybe help with development because it looks very promising.

**Comment by viktorgino** [comment]
Im working on moving the instructions to the Wiki and updating the at the same time, should make the documentation more up-to-date in the future. 

The correct command for the protoc build is : 
`protoc --proto_path=modules/android-auto/headunit/hu/ --cpp_out=modules/android-auto/headunit/hu/generated.x64/ modules/android-auto/headunit/hu/hu.proto`

You'll need to install the dev files for BluezQt. I've already added an updated dependency list to the wiki, I suggest you install those on Ubuntu or Debian : https://github.com/viktorgino/headunit-desktop/wiki/Compiling-HUD

**Comment by pawel0987** [comment]
Oh... Somehow I've missed wiki of this repo...
Thank you for the answer. protoc command you gave works so now I need to deal with installation of dependencies that are missing in ubuntu repository and I hope I will be able to build this monster :)

**Comment by viktorgino** [comment]
@pawel0987 let me know how it goes

**Comment by ExperienceTesla** [comment]
The protoc command work well but i cant do "make -j4 sub-app-pro"
The output is
Make: *** no rule to make target 'sub-app-pro'. Stop.
Same with sub-headunit-desktop-pro

**Comment by viktorgino** [comment]
@ExperienceTesla the `sub-app-pro` is not needed anymore, so the it becomes `make -j4`

**Comment by viktorgino** [comment]
I've added the updated instructions to https://github.com/viktorgino/headunit-desktop/wiki/Building-Headunit-Desktop if anyone has any issues building after following these instructions please let me know here.

**Comment by pawel0987** [comment]
Hello @viktorgino, sorry I haven't write anything for so long, I just didn't have enough time to sit to this :(
I saw your updated wiki pages:
- Building QtCharts (looks good)
- Building QtGStreamer (I have an error when invoking 'make':
```
/usr/include/gstreamer-1.0/gst/gstminiobject.h:33:65: error: invalid cast from type ‘const CapsPtr’ {aka ‘const QGlib::RefPointer<QGst::Caps>’} to type ‘GstMiniObject*’ {aka ‘_GstMiniObject*’}
 #define GST_MINI_OBJECT_CAST(obj)          ((GstMiniObject*)(obj))
                                                                 ^
```
- qtpim - I have found the [repository](https://github.com/qt/qtpim) but I have no idea how to install it :/ Probably because I am not yet in QT or C++ as deep as I should be to work with this project :D 
- Building qofono (it exists in Ubuntu 19.04 - version 0.90 vs 0.96 manually installed, so I am not sure if it is ok to install this with apt or not...)

I will try to deal with this now. If you have any clue what I am doing wrong, please help :)

**Comment by pawel0987** [comment]
The weird thing is that I was able to build headunit app now...
Music player is not working / or I don't know how to use it (could be connected to lack of qtgstreamer I guess). Also the most important tab - Settings - is empty :/

**Comment by pawel0987** [comment]
Update about qtgstreamer:
In ubuntu 19.04 there is package: "libqt5gstreamer-dev" available.
Nevertheless even after installing this and rebuilding application, cannot really use settings view.
May it be something wrong with building process?

**Comment by pawel0987** [comment]
Ok, some summary of what I did and another problems:
Application logs says:
1. `qrc:/qml/SettingsPage/SettingsPageItem.qml:27:9: "Connections.enabled" is not available in QtQuick 2.6.`
Error dissapped when I removed this property.
2. `Settings Loader: ""  : Error loading settings JSON, root type should be of "items" type`
Looks like the cause of empty settings page. I didn't find solution yet.
3. 
    ```
    qrc:/PhoneBluetooth/Phone.qml:251:25: Type Contacts unavailable
    qrc:/PhoneBluetooth/Contacts.qml:2:1: module "QtContacts" is not installed
    ```
I was searching for this module how to install it but without success. It should be in apt but it's not.

One last thing from my site: On [this](https://github.com/viktorgino/headunit-desktop/wiki/Building-Headunit-Desktop#installing-dependencies-debian-based-systems) wiki page I would suggest to not split command into lines. Currently when you copy-paste from wiki it tries to execute separate lines (which is bad of course).
